src/mpi/misc/utils.c

```diff
@@ -163,6 +163,11 @@ static int do_localcopy(const void *sendbuf, MPI_Aint sendcount, MPI_Datatype se
         } else if (send_attr.type == MPL_GPU_POINTER_DEV || recv_attr.type == MPL_GPU_POINTER_DEV) {
             MPL_gpu_free_host(buf);
         }
+
+        if (localcopy_kind == LOCALCOPY_NONBLOCKING) {
```

Comment:
How about move this to the beginning of the branch (e.g. line 106)? By the time we read to the end of the branch, we already lost the context :)

Reply:
Removed this commit since it was added in #6841